Chap_Terms.tex

```diff
@@ -4,39 +4,30 @@
+\item \declareterm{slot}\declareterm{slots}\emph{slot} refers to an allocated set of resources assigned to a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
+\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or cooperatively at any given time during their execution.
+\item \declareterm{job}\emph{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a \emph{session}. For example, ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' is considered a single \ac{MPMD} job containing two applications.
 \item \declareterm{namespace}\emph{namespace} refers to a character string value assigned by the \ac{RM} to a \textit{job}. All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM}.
-\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or in parallel at any given time during their execution.
-\item \declareterm{rank}\emph{rank} refers to the numerical location (starting from zero) of a process within the defined scope. Thus, {global rank} is the rank of a process within its \emph{job}, while \emph{application rank} is the rank of that process within its \emph{application}.
-\item \declareterm{workflow}\declareterm{workflows}\emph{workflow} refers to an orchestrated execution plan frequently spanning multiple \emph{jobs} carried out under the control of a \emph{workflow manager} process. An example workflow might first execute a computational job to generate the flow of liquid through a complex cavity, followed by a visualization job that takes the output of the first job as its input to produce an image output.
+\item \declareterm{rank}\emph{rank} refers to the ordinal number (starting from zero) assigned to a process within a set of peer processes.  For example, the set of processes could be the members of a job, a compute node, or an application execution. \emph{Global rank} is the rank of a process within its \emph{job}, while \emph{application rank} is the rank of that process within its \emph{application}.
```

Comment:
A process is not a member of a node - for one the terminology doesn't match IMHO, but also a node is a piece of HW, which can have multiple OS instances, but there can also be (and there have been systems), which run a global OS across nodes and allow processes to migrate.

Reply:
To the best of my knowledge that type of migration is not well-articulated int eh standard, though should be as the dynamics working group is thinking along those lines. Maybe a following PR developed with that working group could suggest some language here that is sufficiently flexible?

Reply:
That's not quite what I meant: a node is not well defined term for a piece of hardware, not a logical instance you schedule something to. Assuming we mean OS processes, we schedule them to OS instances. and then OS has control of them (to some degree). There can be one OS instance spanning several nodes, and there can be several OS instances on one node. IMHO we should not use node here, since different architectures have different meanings for it (and the meaning is changing with the increased use of super nodes (e.g., DGXs).

Reply:
"the set of processes could be the members of a job, a compute node, or an application execution" Is this what you mean Martin? The way this is written it implies "the set of processes could be members of a compute node." I think that it would be better and more clear as:

"the set of processes could be the members of a job, executing on the same compute node, or part of an application execution."